### PR TITLE
#4344: Update domain manager invite/removal emails to include org admin clarifier

### LINKED ISSUE #4344: Dev: Add text to notification emails to identify org admins when they take action
### Issue description

Update email content so that actions taken by org admins—that may cause user confusion—are clearly identified as such.

Add "(Org admin)" to email notification to clarify where changes are coming from.

### Acceptance criteria

- [ ] Update the following emails to indicate when an action is initiated by an org admin.
  - [ ] [Domain manager invited (inform existing managers)](https://docs.google.com/document/d/18eQRA5QWu5Ct6ICK2-FCZJf9kbbk4LklekRRfb6g9tQ/edit?tab=t.tdy28x39shmc#heading=h.cvnj3a9w6chc)
  - [ ] [Domain manager removed (inform existing managers)](https://docs.google.com/document/d/18eQRA5QWu5Ct6ICK2-FCZJf9kbbk4LklekRRfb6g9tQ/edit?tab=t.xb97t9vc641x#heading=h.gjdgxs)

### Additional context

_No response_

### Links to other issues

- 🔄 #4150

### PR DESCRIPTION
**Pull request by smishraECS**
## Ticket

Resolves #4344

## Changes

Updated email templates to add "(Org admin)" clarification tag

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Update documentation in READMEs and/or onboarding guide

#### Ensured code standards are met (Original Developer)

- [ ] If any updated dependencies on Pipfile, also update dependencies in requirements.txt.
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] [Follow the process for requesting a design review](https://dhscisa.enterprise.slack.com/docs/T02QH7E1MHA/F06CZ6MRUKA). If code is not user-facing, delete design reviewer checklist
- [ ] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Reviewed [accessibility checklist](https://github.com/cisagov/manage.get.gov/blob/main/docs/dev-practices/code_review.md#accessibility-checklist) using screen reader (such as NVDA with Chrome or Voiceover with Safari), ANDI, or WAVE:
  - [ ] Tested general usability
  - [ ] Page header structure
  - [ ] Landmarks
  - [ ] Links and buttons
- [ ] Checked for errors or warnings with an a11y browser tool (such as ANDI or WAVE)

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Verified code meets all checks above. Address any checks that are not satisfied
- [ ] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [ ] Checked that all code is adequately covered by tests
- [ ] Verify migrations are valid and do not conflict with existing migrations

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A.

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Meets all designs and user flows provided by design/product
- [ ] Checked keyboard navigability
- [ ] Reviewed [accessibility checklist](https://github.com/cisagov/manage.get.gov/blob/main/docs/dev-practices/code_review.md#accessibility-checklist) using screen reader (such as NVDA with Chrome or Voiceover with Safari), ANDI, or WAVE:
  - [ ] Tested general usability
  - [ ] Page header structure
  - [ ] Landmarks
  - [ ] Links and buttons
- [ ] Checked for errors or warnings with an a11y browser tool (such as ANDI or WAVE)
- [ ] (Rarely needed) Tested as both an analyst and applicant user

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior. Comment any found issues or broken flows.
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Reviewed [accessibility checklist](https://github.com/cisagov/manage.get.gov/blob/main/docs/dev-practices/code_review.md#accessibility-checklist) using screen reader (such as NVDA with Chrome or Voiceover with Safari), ANDI, or WAVE:
  - [ ] Tested general usability
  - [ ] Page header structure
  - [ ] Landmarks
  - [ ] Links and buttons
- [ ] Checked for errors or warnings with an a11y browser tool (such as ANDI or WAVE)
- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### References
- [Code review best practices](../docs/dev-practices/code_review.md)

### CODE REVIEW

**Review by allly-b — CHANGES_REQUESTED** [review]
Hey, it looks like `send_domain_manager_removal_emails_to_domain_managers` is called on the `DomainDeleteUserView` which allows for admins and domain managers to see the deletion page (this is controlled by this code: `@grant_access(IS_DOMAIN_MANAGER, IS_STAFF_MANAGING_DOMAIN) `). Likewise, the view that controls sending the invitation is accessible by domain managers in general, who may not be the org admin.

Your code needs to only add the org admin text if the user actually is an org admin, you will likely need to create a new variable that stores "is user an org admin", then pass it the send email functions. You should be able to have an if statement in these templates then that conditionally shows org admin. There's several other ways to do it though. 

Let me know if you want to huddle on this, or if you have follow up questions.

File: src/registrar/utility/email_invitations.py [hunk 1]
```diff
@@ -60,6 +60,27 @@ def _check_outside_org_membership(email, requestor, is_member_of_different_org):
...
+    Returns:
+        Boolean indicating if user is an Org Admin.
+    """
+
+    # Check to see if the user is an Org Admin
+    roles_list = list(requestor.get_portfolios())
+    if not roles_list:
+        return False
+    for role in roles_list:
+        if UserPortfolioRoleChoices.ORGANIZATION_ADMIN in role.roles:
```
Comment (allly-b) [inline src/registrar/utility/email_invitations.py]:
```suggestion
def _check_user_org_admin(requestor, portfolio) -> bool:
    """
    Check to see if the requestor is an org admin of the current portfolio

    Args:
        requestor (User): The user initiating the invitation.
        portfolio (Portfolio): The portfolio the invitation is associated with

    Returns:
        Boolean indicating if user is an Org Admin.
    """

    # Check to see if the user is an Org Admin of the current portfolio
        return UserPortfolioPermission.objects.filter(user=requestor, roles__contain=[UserPortfolioRoleChoices.ORGANIZATION_ADMIN], portfolio=portfolio).exists()
```


Thanks for writing such great comments! 

One caveat that was definitely not explicitly mentioned in the ticket context is that someone can be an org admin in one portfolio and be just a basic member on another portfolio. The code was right and was just missing a filter to ensure that the org admin is org admin on the **current** portfolio. 

Also, I suggest using just a filter on `UserPortfolioPermissions` to stay in line with what is done elsewhere [in this file](https://github.com/cisagov/manage.get.gov/blob/1cf28ff57db46535297cd586db32a41200aeb79c/src/registrar/utility/email_invitations.py#L324). However, you do not have to follow this format if you don't want to. It's the part about adding a filter on portfolio that's ultra important here.

File: src/registrar/tests/test_email_invitations.py [hunk 2]
```diff
@@ -1235,6 +1269,8 @@ def test_send_email_failure(self, mock_filter, mock_send_templated_email):
         """Test handling of failure in sending admin removal emails."""
         mock_filter.return_value.exclude.return_value = [self.domain_manager1, self.domain_manager2]
 
+        # mock_check_user_org_admin.return_value = False
```
Comment (allly-b) [inline src/registrar/tests/test_email_invitations.py]:
```suggestion
```
**Nitpick:** delete

File: src/registrar/utility/email_invitations.py [hunk 3]
```diff
@@ -131,7 +152,8 @@ def send_domain_invitation_email(
     """
     domains = _normalize_domains(domains)
     requestor_email = _get_requestor_email(requestor, domains=domains)
-
+    # Check to see if the user sending the invitation is an Org Admin
+    is_org_admin = _check_user_org_admin(requestor)
```
Comment (allly-b) [inline src/registrar/utility/email_invitations.py]: following on from the previous comment, here you should get portfolio from the domains variable.

**Review by allly-b — CHANGES_REQUESTED** [review]
One blocking change request, it's checking if the user is an org admin, and just needs to add in there a check that the user is an org admin on the  _current portfolio._ Right now this email template is used if the user is an org admin on any portfolio.

**Comment by Katherine-MN** [comment]
@smishraECS This looks good; however, I noticed the same issue @allly-b noted above. I'm part of two portfolios, one with admin permissions and the other with basic permissions. When I made a change in the portfolio where I had basic permissions, the email indicated I was an "org admin" likely because I have that permission for the other portfolio. Once you adjust that issue, I can test again and complete my review!

**Comment by smishraECS** [comment]
@Katherine-MN @allly-b
Thanks for the review, didn't catch think of those edge case scenarios!

Issues should be resolved with this latest commit

**Review by Katherine-MN — APPROVED** [review]
LGTM!

**Review by allly-b — APPROVED** [review]
LGTM thanks for making the changes
